nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManagerComponent.java

```diff
@@ -361,12 +371,18 @@ protected String stripBlobKeyPrefix(String key) {
     // find which GCs to use
     // only GC the binary managers to which we dispatch blobs
     protected List<BinaryGarbageCollector> getGarbageCollectors() {
-        List<BinaryGarbageCollector> gcs = new LinkedList<>();
-        for (String providerId : getBlobDispatcher().getBlobProviderIds()) {
-            BlobProvider blobProvider = getBlobProvider(providerId);
-            BinaryGarbageCollector gc = blobProvider.getBinaryGarbageCollector();
-            if (gc != null) {
-                gcs.add(gc);
+        if (gcs == null) {
+            synchronized (this) {
+                if (gcs == null) {
+                    gcs = new LinkedList<>();
```

Comment:
For the double-checked locking idiom you want to assign `gcs` only once and at then _end_, when the value is completely determined. So 

    List<BinaryGarbageCollector> list = new ArrayList<>(); // btw we don't use LinkedList 
    ...
        if (gc != null) { list.add(gc); }
    ...
    gcs = list;